druid-filtering-ext/src/main/java/bi/deep/filtering/common/IPRange.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package bi.deep.filtering.common;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import inet.ipaddr.IPAddress;
+import inet.ipaddr.IPAddressString;
+import inet.ipaddr.format.IPAddressRange;
+import java.util.Objects;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.druid.error.InvalidInput;
+import org.apache.druid.java.util.common.IAE;
+
+public class IPRange {
+    private static final String SEPARATOR = "/";
+    private final IPAddressRange addressRange;
+    private final IPAddress.IPVersion ipVersion;
+
+    @JsonCreator
+    public IPRange(String range) {
+        if (StringUtils.isEmpty(range)) {
+            throw InvalidInput.exception("Range cannot be null or empty");
+        }
+
+        final String[] parts = range.split(SEPARATOR);
+
+        if (parts.length != 2) {
+            throw InvalidInput.exception(
+                    String.format("Range should include lower and upper bounds in format `lower%supper`", SEPARATOR));
+        }
+
+        final IPAddress lower = new IPAddressString(parts[0]).getAddress();
+        final IPAddress upper = new IPAddressString(parts[1]).getAddress();
+
+        if (lower == null) {
+            throw InvalidInput.exception("Invalid lower IP address");
+        }
+
+        if (upper == null) {
+            throw InvalidInput.exception("Invalid upper IP address");
+        }
+
+        if (lower.getIPVersion() != upper.getIPVersion()) {
```

Comment:
```suggestion
       if (lower.getIPVersion().equals(upper.getIPVersion()))
```

Reply:
It is an enum, so equality will be same as equal

Reply:
Yeah, it'll just look a bit prettier this way, nothing else :)